docs/async.md

```diff
@@ -0,0 +1,210 @@
+# Async Client
+
+HTTPX offers a standard synchronous API by default, but also gives you
+the option of an async client if you need it.
+
+Async is a concurrency model that is far more efficient than multi-threading,
+and can provide significant performance benefits and enable the use of
+long-lived network connections such as WebSockets.
+
+If you're working with an async web framework then you'll also want to use an
+async client for sending outgoing HTTP requests.
+
+## Making Async requests
+
+To make asynchronous requests, you'll need an `AsyncClient`.
+
+```python
+>>> async with httpx.AsyncClient() as client:
+>>>     r = await client.get('https://www.example.com/')
+>>> r
+<Response [200 OK]>
+```
+
+!!! tip
+    Use [IPython](https://ipython.readthedocs.io/en/stable/) to try this code interactively, as it supports executing `async`/`await` expressions in the console.
+
+## API Differences
+
+If you're using streaming responses then there are a few bits of API that
+use async methods.
+
+### Making requests
+
+The request methods are all async, so you should use `response = await client.get(...)` style for all of the following:
+
+* `AsyncClient.get(url, ...)`
+* `AsyncClient.options(url, ...)`
+* `AsyncClient.head(url, ...)`
+* `AsyncClient.post(url, ...)`
+* `AsyncClient.put(url, ...)`
+* `AsyncClient.patch(url, ...)`
+* `AsyncClient.delete(url, ...)`
+* `AsyncClient.request(method, url, ...)`
+* `AsyncClient.send(request, ...)`
+
+### Opening and closing clients
+
+Use `async with httpx.AsyncClient()` if you want a context-managed client...
+
+```python
+async with httpx.AsyncClient() as client:
+    ...
+```
+
+Alternatively, use `await clien.aclose()` if you want to close a client explicitly.
+
+```python
+client = httpx.AsyncClient()
+...
+await client.aclose()
+```
+
+### Streaming responses
+
+The `AsyncClient.stream(method, url, ...)` method is an async context block.
+
+```python
+>>> client = httpx.AsyncClient()
+>>> async with client.stream('GET', 'https://www.example.com/') as response:
+>>>     async for chunk in r.aiter_content():
+>>>         ...
+```
+
+The async response streaming methods are:
+
+* `Response.aread()` - For conditionally reading a response inside a stream block.
+* `Response.aiter_content()` - For streaming the response content as bytes.
+* `Response.aiter_text()` - For streaming the response content as text.
+* `Response.aiter_lines()` - For streaming the response content as lines of text.
+* `Response.aiter_raw()` - For streaming the raw response bytes, without applying content decoding.
+* `Response.aclose()` - For closing the response. You don't usually need this, since `.stream` block close the response automatically on exit.
+
+### Streaming requests
+
+When sending a request body with an `AsyncClient` instance, it is invalid to use a bytes generator, but it is valid to use an async bytes generator...
```

Comment:
```suggestion
When sending a streaming request body with an `AsyncClient` instance, you should use an async butes generator instead of a bytes generator:
```